### ISSUE TITLE
cannot pickle '_thread.RLock' object

### ISSUE THREAD

**Issue by stefanodvx** [pydantic/pydantic#10656]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

Getting the following exception and can't understand where it comes from and how to fix this (`botapi` is my own package, you can check the [code](https://github.com/uvdbot/botapi) here):
```bash
Traceback (most recent call last):
  File "/app/main.py", line 5, in <module>
    from botapi.api import BotAPI
  File "/usr/local/lib/python3.11/site-packages/botapi/__init__.py", line 1, in <module>
    from .api import BotAPI
  File "/usr/local/lib/python3.11/site-packages/botapi/api.py", line 16, in <module>
    class BotAPI(BaseModel, Methods):
  File "/usr/local/lib/python3.11/site-packages/pydantic/_internal/_model_construction.py", line 224, in __new__
    complete_model_class(
  File "/usr/local/lib/python3.11/site-packages/pydantic/_internal/_model_construction.py", line 610, in complete_model_class
    generate_pydantic_signature(init=cls.__init__, fields=cls.model_fields, config_wrapper=config_wrapper),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pydantic/_internal/_signature.py", line 159, in generate_pydantic_signature
    merged_params = _generate_signature_parameters(init, fields, config_wrapper)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pydantic/_internal/_signature.py", line 115, in _generate_signature_parameters
    kwargs = {} if field.is_required() else {'default': field.get_default(call_default_factory=False)}
                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pydantic/fields.py", line 566, in get_default
    return _utils.smart_deepcopy(self.default)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pydantic/_internal/_utils.py", line 319, in smart_deepcopy
    return deepcopy(obj)  # slowest way when we actually might need a deepcopy
           ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/copy.py", line 172, in deepcopy
    y = _reconstruct(x, memo, *rv)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/copy.py", line 271, in _reconstruct
    state = deepcopy(state, memo)
            ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/copy.py", line 146, in deepcopy
    y = copier(x, memo)
        ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/copy.py", line 231, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
                             ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/copy.py", line 172, in deepcopy
    y = _reconstruct(x, memo, *rv)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/copy.py", line 271, in _reconstruct
    state = deepcopy(state, memo)
            ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/copy.py", line 146, in deepcopy
    y = copier(x, memo)
        ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/copy.py", line 231, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
                             ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/copy.py", line 172, in deepcopy
    y = _reconstruct(x, memo, *rv)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/copy.py", line 271, in _reconstruct
    state = deepcopy(state, memo)
            ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/copy.py", line 146, in deepcopy
    y = copier(x, memo)
        ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/copy.py", line 231, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
                             ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/copy.py", line 161, in deepcopy
    rv = reductor(4)
         ^^^^^^^^^^^
TypeError: cannot pickle '_thread.RLock' object
```

### Example Code

```Python
class BotAPI(BaseModel, Methods):
    token: str
    api_url: str = "https://api.telegram.org"
    parse_mode: str = "HTML"
    session: httpx.AsyncClient = httpx.AsyncClient(timeout=120)
    sudoers: List[int] = Field(default_factory=list)

    class Config:
        arbitrary_types_allowed = True
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.9.2
        pydantic-core version: 2.23.4
          pydantic-core build: profile=release pgo=false
                 install path: /home/ste/.local/lib/python3.11/site-packages/pydantic
               python version: 3.11.10 (main, Sep  7 2024, 18:35:42) [GCC 9.4.0]
                     platform: Linux-5.4.0-186-generic-x86_64-with-glibc2.31
             related packages: typing_extensions-4.12.2 fastapi-0.115.2
                       commit: unknown
```

**Comment by stefanodvx** [comment]
More info:
Can confirm that when removing `httpx.AsyncClient(timeout=120)` and putting any other thing, like `None`, the error disappear. So I guess it's a problem with httpx and pydantic... How should I fix it?

**Comment by Viicos** [comment]
This will be fixed in the next 2.10 release.
